base/boot.jl

```diff
@@ -123,7 +123,7 @@ export
     SimpleVector, AbstractArray, DenseArray, NamedTuple,
     # special objects
     Function, CodeInfo, Method, MethodTable, TypeMapEntry, TypeMapLevel,
-    Module, Symbol, Task, Array, WeakRef, VecElement,
+    Module, Symbol, Task, Array, Uninitialized, uninitialized, WeakRef, VecElement,
```

Comment:
Should `Uninitialized` be exported? It does not seem necessary for the usage

Reply:
`Uninitialized` must be exported, lest Julia not build :).

Reply:
Why is that?

Reply:
`Uninitialized`/`uninitialized` must be visible everywhere uninitialized arrays are or will be constructed; these exports make `Uninitialized`/`uninitialized` visible beyond `Core`. The build failure absent these exports comes from the definitions using `Uninitialized`/`uninitialized` in `sysimg.jl`. Does that clarify? :)

Reply:
Though wouldn't it be possible to export only `uninitialized` from Base (not Core)? Should `Uninitialized` be used outside of Base?

Reply:
`Uninitialized` should see use outside Base, yes, in definitions of constructors for uninitialized arrays generally (see #24595). Additionally `uninitialized` is merely an alias for `Uninitialized()`, and hence `uninitialized` cannot do its magic without `Uninitialized` visible. Does that clarify? :)